### ISSUE TITLE
YouTube music videos not playing

### ISSUE THREAD

**Issue by Starduck536** [Tsunami014/Grayjay-plugins#8]
Whenever I heard of playing a YouTube music video Fail to load video plugin YouTube music. YouTube music cannot use in operator to search for URL is undefined. 140 14-15

**Comment by Tsunami014** [comment]
I see. I'll look into it whenever I can next

**Comment by Tsunami014** [comment]
I figured out the issue, and it's that it requires login. I just made a change to throw an exception when it requires login, but that's kinda weird as you can play it on the website without login. Maybe google finally figured it out and I'm going to need to change the whole code to do web scraping.

I'll give the web scraping a shot (which requires much work, this'll take a while), but I think I'll keep in the API just in case

**Comment by anuragtyagi-dev** [comment]
Issue is still there when done Login with Gmail Account for YT Music

**Comment by Tsunami014** [comment]
Yeah sorry I haven't been able to work on it recently and I'm kinda on holidays but I'll try to fix it soon!

**Comment by anuragtyagi-dev** [comment]
Thanks for helping the community.

**Comment by Tsunami014** [comment]
<3 tsym!

Just got back. There's been much procrastination (sorry) (I mean, dealing with Google changing something that was already complex and obfuscated is very annoying) and issues with connecting to mobile (which fueled the procrastination) (I kept saying it wasn't the firewall, it *can't* be; but it was), then holiday, and now after 1/2 a day of working on it (the other half fixing the connection issue), I've managed to half fix the problem. 🎉

I'll try working more on it but in the very limited time I've had so far fixing the next issue it seems really weird, and I have nothing on how to fix at the moment. But as I said, only spend about 2 mins looking into, maybe can find out something tomorrow?

Current issue is now: it's gaining the URL from Google, but when you follow the url it doesn't show the video >:(

**Comment by Tsunami014** [comment]
I got lead down a large rabbit hole of 'how to reverse engineer YouTube's API' where I wrote some cool as code that still didn't fix the problem. I did find out that it's getting a 403 forbidden response, but I have no clue what YT did or how to make it work again :(
Everything I try gives the same result; a link leading to a blank page, with a 403 error in the console. That guide was the 2nd edition but I tried some of the things in it and got different results, so methinks YT did something after that was made.
I *really* don't know how to fix this.

**Comment by Tsunami014** [comment]
Actually, I believe I have figured out what the problem is.

According to [other issues](https://github.com/futo-org/grayjay-android/issues/1815#issuecomment-2599660382), it seems as though YouTube is trying it's hardest to stop simple IOS (& other) streams via ip blocking (or [something similar](https://github.com/futo-org/grayjay-android/issues/1639#issuecomment-2605430623)) BUT nevertheless, it means that ways that worked before now won't. You'll notice that the regular YT plugin also is having problems, and that's because the UMP streams Grayjay has emplored as a solution [is working less and less](https://www.reddit.com/r/grayjay/comments/1i6u2o6/ump_streams_the_issue_we_all_seem_to_have_rn/) and now we're all a bit stuck.

[They're trying their hardest to fix it](https://github.com/futo-org/grayjay-android/issues/1815#issuecomment-2603148219), but again; it's only a matter of time. Someone once said it's like a game of cat and mouse; one side trying to break it as fast as possible and the other trying to fix it.

So, all we can do rn is wait.
